samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java

```diff
@@ -265,23 +272,34 @@ private static void findReachableJoins(OperatorSpec operatorSpec, StreamEdge sou
     }
   }
 
-  private static void calculateIntStreamPartitions(JobGraph jobGraph, Config config) {
-    int partitions = config.getInt(JobConfig.JOB_INTERMEDIATE_STREAM_PARTITIONS(), StreamEdge.PARTITIONS_UNKNOWN);
-    if (partitions < 0) {
+  private static void calculateIntermediateStreamPartitions(JobGraph jobGraph, Config config) {
+    final String defaultPartitionsConfigProperty = JobConfig.JOB_INTERMEDIATE_STREAM_PARTITIONS();
+    int partitions = config.getInt(defaultPartitionsConfigProperty, StreamEdge.PARTITIONS_UNKNOWN);
+    if (partitions == StreamEdge.PARTITIONS_UNKNOWN) {
       // use the following simple algo to figure out the partitions
       // partition = MAX(MAX(Input topic partitions), MAX(Output topic partitions))
       // partition will be further bounded by MAX_INFERRED_PARTITIONS.
       // This is important when running in hadoop where an HDFS input can have lots of files (partitions).
-      int maxInPartitions = maxPartition(jobGraph.getSources());
-      int maxOutPartitions = maxPartition(jobGraph.getSinks());
+      int maxInPartitions = maxPartitions(jobGraph.getInputStreams());
+      int maxOutPartitions = maxPartitions(jobGraph.getOutputStreams());
       partitions = Math.max(maxInPartitions, maxOutPartitions);
 
       if (partitions > MAX_INFERRED_PARTITIONS) {
         partitions = MAX_INFERRED_PARTITIONS;
         log.warn(String.format("Inferred intermediate stream partition count %d is greater than the max %d. Using the max.",
             partitions, MAX_INFERRED_PARTITIONS));
       }
+    } else {
+      // Reject any zero or other negative values explicitly specified in config.
```

Comment:
For my own understanding, can you provide an example of when this code-path would be reached?

Reply:
Someone mistakenly setting `job.intermediate.stream.partitions` to 0 or -2

Reply:
thanks for adding the validation!

Reply:
My pleasure (^_^;)